ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/genesis/GenesisHandler.java

```diff
@@ -51,42 +44,33 @@ public void onDepositsFromBlock(final DepositsFromBlockEvent event) {
     if (!chainStorageClient.isPreGenesis()) {
       return;
     }
-
-    final Bytes32 eth1BlockHash = event.getBlockHash();
-    final UnsignedLong eth1Timestamp = event.getBlockTimestamp();
     final List<DepositWithIndex> deposits =
         event.getDeposits().stream()
             .map(DepositUtil::convertDepositEventToOperationDeposit)
             .collect(Collectors.toList());
-    genesisGenerator.addDepositsFromBlock(eth1BlockHash, eth1Timestamp, deposits);
 
-    if (config.getDepositMode().equals(Constants.DEPOSIT_SIM)) {
-      genesisGenerator
-          .getGenesisStateIfValid(BeaconStateUtil::is_valid_genesis_stateSim)
-          .ifPresent(
-              candidate_state -> {
-                setSimulationGenesisTime(candidate_state);
-                eth2Genesis(new GenesisEvent(candidate_state));
-              });
-    } else {
-      genesisGenerator
-          .getGenesisStateIfValid(BeaconStateUtil::is_valid_genesis_state)
-          .ifPresent(candidate_state -> eth2Genesis(new GenesisEvent(candidate_state)));
-    }
+    processNewData(event.getBlockHash(), event.getBlockTimestamp(), deposits);
+  }
+
+  @Override
+  public void onMinGenesisTimeBlock(MinGenesisTimeBlockEvent event) {
+    processNewData(event.getBlockHash(), event.getTimestamp(), List.of());
   }
 
-  private void eth2Genesis(GenesisEvent genesisEvent) {
+  private void processNewData(
+      Bytes32 blockHash, UnsignedLong timestamp, List<DepositWithIndex> deposits) {
+    genesisGenerator.updateCandidateState(blockHash, timestamp, List.of());
```

Comment:
We should actually use the deposits list passed in.
```suggestion
    genesisGenerator.updateCandidateState(blockHash, timestamp, deposits);
```

Reply:
done.